src/GoTrueClient.ts

```diff
@@ -1004,24 +1037,31 @@ export default class GoTrueClient {
     return isValidSession
   }
 
-  private _handleProviderSignIn(
+  private async _handleProviderSignIn(
     provider: Provider,
     options: {
       redirectTo?: string
       scopes?: string
       queryParams?: { [key: string]: string }
       skipBrowserRedirect?: boolean
+      flowType?: OAuthFlowType
     } = {}
   ) {
-    const url: string = this._getUrlForProvider(provider, {
+    if (!options.flowType) {
+      options.flowType = 'implicit'
+    }
```

Comment:
```suggestion
```

Reply:
can remove this if you choose to set the default value of `flowType` in the suggestion i made earlier for `signInWithOAuth`